### PR TITLE
feat(ingestion) Add new endpoint to test an ingestion connection

### PR DESCRIPTION
**Pull request by chriscollins3456**
Adds a new GraphQL endpoint that allows the user to test an ingestion connection given a recipe. This is only the backend piece, but it will be used by the frontend in order to allow users building new ingestion sources through UI Managed Ingestion to ensure their connection is set up properly.

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-graphql-core/src/main/resources/ingestion.graphql [hunk 1]
```diff
@@ -66,6 +66,12 @@ extend type Mutation {
   Cancel a running execution request, provided the urn of the original execution request
   """
   cancelIngestionExecutionRequest(input: CancelIngestionExecutionRequestInput!): String
+
+  """
+  Create a request to execute a test ingestion connection job
+  input: Input required for creating a test connection request
+  """
+  createTestConnectionRequest(input: CreateTestConnectionRequestInput!): String
```
Comment (aditya-radhakrishnan) [inline datahub-graphql-core/src/main/resources/ingestion.graphql:74]: I'm a bit confused - why is the return type here a String? Would a Boolean not be better (connection works or does not)? I know we return Strings from some of our GraphQL endpoints but I am personally not a huge fan

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/ingest/execution/CreateTestConnectionRequestResolver.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    _ingestionConfiguration = ingestionConfiguration;
+  }
+
+  @Override
+  public CompletableFuture<String> get(final DataFetchingEnvironment environment) throws Exception {
+    final QueryContext context = environment.getContext();
+
+    return CompletableFuture.supplyAsync(() -> {
+
+      if (IngestionAuthUtils.canManageIngestion(context)) {
```
[the remark this answers is not in the document]
Comment (chriscollins3456) [reply datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/ingest/execution/CreateTestConnectionRequestResolver.java:52]: sounds good!

File: datahub-graphql-core/src/main/resources/ingestion.graphql (hunk 1 above)
Comment (jjoyce0510) [reply datahub-graphql-core/src/main/resources/ingestion.graphql:74]: So here we return an URN for the task. 

Then we poll for the result of that urn... It's not blocking

**Review by jjoyce0510 — APPROVED** [review]
This looks great to me overall. I'm guessing the next thing is an API to check the result?

**Pull request merged**